libs/pipeline.py

```diff
@@ -0,0 +1,130 @@
+"""
+Code that is used to help move information around in the pipeline, starting with `Region` which
+represents a geographical area (state, county, metro area, etc).
+"""
+
+
+import json
+import os
+from typing import Optional
+
+import pandas as pd
+import structlog
+from pydantic import BaseModel
+
+import pyseir
+from covidactnow.datapublic.common_fields import CommonFields
+from libs.datasets import FIPSPopulation
+from libs.datasets import combined_datasets
+from pyseir.rt.utils import NEW_ORLEANS_FIPS
+from pyseir.utils import get_run_artifact_path, RunArtifact
+
+
+_log = structlog.get_logger()
+
+
+class Region(BaseModel):
+    """Identifies a geographical area."""
+
+    # The FIPS identifier for the region, either 2 digits for a state or 5 digits for a county.
+    # TODO(tom): Add support for regions other than states and counties.
+    fips: str
+
+    def __repr__(self) -> str:
+        return f"Region(fips={self.fips})"
+
+    @staticmethod
+    def from_fips(fips: str) -> "Region":
+        return Region(fips=fips)
+
+
+class RegionalCombinedData(BaseModel):
+    """Identifies a geographical area and wraps access to `combined_datasets` of it."""
+
+    region: Region
+
+    @staticmethod
+    def from_fips(fips: str) -> "RegionalCombinedData":
+        return RegionalCombinedData(region=Region.from_fips(fips))
+
+    def get_us_latest(self):
+        """Gets latest values for a given state or county fips code."""
+        us_latest = combined_datasets.load_us_latest_dataset()
+        return us_latest.get_record_for_fips(self.region.fips)
+
+    def get_population(self) -> int:
+        """Gets the population for this region."""
+        return self.get_us_latest()[CommonFields.POPULATION]
+
+
+class RegionalApiInput(BaseModel):
```

Comment:
What is the intent of Api Input?  just worried that API is a bit of an overloaded term

Reply:
Changed to `RegionalWebUIInput`